R/pkg/R/sparkR.R

```diff
@@ -344,6 +344,7 @@ sparkRHive.init <- function(jsc = NULL) {
 #' @note sparkR.session since 2.0.0
 sparkR.session <- function(
   master = "",
+  deployMode = "",
```

Comment:
Hmm, I think the standard way to do that in Spark is to have the deploy mode config:
YARN cluster
master=yarn
deploy-mode=cluster

YARN client
master=yarn
deploy-mode=client

deploy-mode is default to client.

In this as on the SparkSession API does it make sense to support deploy-mode cluster though? When running in the YARN cluster mode the driver JVM starts and then its companion R process. The R SparkSession API could not be able to change the master or the deploy-mode at the point R code is running? (Unless we want to support a remote R process connecting to the Spark driver running in the YARN cluster - another JIRA on this)

In any other case only master is used (standalone, mesos).

I think now one could still set spark.deployMode in the named list (though again won't be effective).

Perhaps we just document this in the programming guide regrading how to use the SparkSession R API with YARN?